gen/ldap_it4i

```diff
@@ -0,0 +1,105 @@
+our $A_FIRST_NAME;  *A_FIRST_NAME = \'urn:perun:user:attribute-def:core:firstName';
+our $A_LAST_NAME;  *A_LAST_NAME = \'urn:perun:user:attribute-def:core:lastName';
+our $A_DISPLAY_NAME;  *A_DISPLAY_NAME = \'urn:perun:user:attribute-def:core:displayName';
+our $A_USER_PREFERRED_MAIL;  *A_USER_PREFERRED_MAIL = \'urn:perun:user:attribute-def:def:preferredMail';
+our $A_SSHKEYS;  *A_SSHKEYS = \'urn:perun:user:attribute-def:def:sshPublicKey';
+our $A_MEMBER_STATUS;  *A_MEMBER_STATUS =  \'urn:perun:member:attribute-def:core:status';
+
+our $STATUS_VALID;      *STATUS_VALID =        \'VALID';
+our $STATUS_EXPIRED;    *STATUS_EXPIRED =      \'EXPIRED';
+our $STATUS_INVALID;    *STATUS_INVALID =      \'INVALID';
+our $STATUS_DISABLED;   *STATUS_DISABLED =     \'DISABLED';
+
+# check facility attribute
+my $ldapBaseDN = $data->getFacilityAttributeValue( attrName => $A_F_BASE_DN );
+if (!defined($ldapBaseDN)) {
+	exit 1;
+}
+
+# gather user data
+my $users;
+
+foreach my $resourceId ( $data->getResourceIds() ) {
+	foreach my $memberId ( $data->getMemberIdsForResource( resource => $resourceId )) {
+
+		my $login = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN_EINFRA );
+		$users->{$login}->{$A_FIRST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_FIRST_NAME );
+		$users->{$login}->{$A_LAST_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_LAST_NAME );
+		$users->{$login}->{$A_DISPLAY_NAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_DISPLAY_NAME );
+		$users->{$login}->{$A_USER_PREFERRED_MAIL} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_PREFERRED_MAIL );
+		$users->{$login}->{$A_SSHKEYS} = $data->getUserAttributeValue( member => $memberId, attrName => $A_SSHKEYS );
+
+		my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_STATUS );
+		if ($status eq $STATUS_VALID) {
+			$status = 'ACTIVE';
+		} elsif ($status eq $STATUS_INVALID) {
+			$status = 'INCOMPLETE';
+		} elsif ($status eq $STATUS_EXPIRED) {
+			$status = 'INACTIVE';
+		} elsif ($status eq $STATUS_DISABLED) {
+			$status = 'ARCHIVED';
+		}
+		$users->{$login}->{$A_MEMBER_STATUS} = $status;
+	}
+}
+
+# print BASE_DN file
+open FILE,">:encoding(UTF-8)","$baseDnFileName" or die "Cannot open $baseDnFileName: $! \n";
+print FILE $ldapBaseDN;
+close(FILE);
+
+# print user data LDIF
+open FILE,">:encoding(UTF-8)","$fileName" or die "Cannot open $fileName: $! \n";
+
+for my $login (sort keys %$users) {
+
+	print FILE "dn: uid=" . $login . "," . $ldapBaseDN . "\n";
+	print FILE "uid: " . $login . "\n";
+	print FILE "givenName: " . $users->{$login}->{$A_FIRST_NAME} . "\n";
+	print FILE "sn: " . $users->{$login}->{$A_LAST_NAME} . "\n";
+	print FILE "fullName: " . $users->{$login}->{$A_FIRST_NAME} . " " . $users->{$login}->{$A_LAST_NAME} . "\n";
```

Comment:
Instead of `fullName` this will be `cn`. But attribute is required by schema, so the script should handle empty names more gracefully. See other services for AD, where value is constructed differently (usually for the displayName) where we check for empty values and if result is empty, its replaced with `N/A` (this is probably only in LDAPc).

Reply:
Added check of `emptiness` of `givenName` and sn values and the cn value is `N/A` if both of them are empty.